src/core/thread/link_metrics.cpp

```diff
@@ -164,37 +173,76 @@ otError LinkMetrics::SendMgmtRequestForwardTrackingSeries(const Ip6::Address &
 
     memcpy(subTlvs + sizeof(Tlv), &aSeriesId, sizeof(aSeriesId));
 
-    seriesFlags->Clear();
-    if (aSeriesFlags.mLinkProbe)
+    seriesFlags->SetFromPublicSeriesFlags(aSeriesFlags);
+
+    error = Get<Mle::MleRouter>().SendLinkMetricsManagementRequest(aDestination, subTlvs,
+                                                                   forwardProbingRegistrationSubTlv->GetSize());
+
+exit:
+    otLogDebgMle("SendMgmtRequestForwardTrackingSeries, error:%s, Series ID:%u", otThreadErrorToString(error),
+                 aSeriesId);
+    return error;
+}
+
+otError LinkMetrics::SendMgmtRequestEnhAckProbing(const Ip6::Address &           aDestination,
+                                                  const otLinkMetricsEnhAckFlags aEnhAckFlags,
+                                                  const otLinkMetrics *          aLinkMetricsFlags)
+{
+    otError                error             = OT_ERROR_NONE;
+    LinkMetricsEnhAckFlags enhAckFlags       = static_cast<LinkMetricsEnhAckFlags>(aEnhAckFlags);
+    uint8_t                typeIdFlagsOffset = sizeof(Tlv) + sizeof(enhAckFlags);
+    uint8_t                typeIdFlagsCount  = 0;
+    uint8_t      subTlvs[sizeof(Tlv) + sizeof(enhAckFlags) + sizeof(LinkMetricsTypeIdFlags) * kMaxTypeIdFlags];
+    Tlv *        enhancedAckLinkMetricsConfigurationSubTlv = reinterpret_cast<Tlv *>(subTlvs);
```

Comment:
Not necessary for this PR, but I think this could be cleaned up by having a class that generates/parses Enhanced ACK Link Metrics Configuration Sub-TLVs.

Reply:
Added a class for generating Enhanced ACK Link Metrics Configuration Sub-TLVs.